app/controllers/devise/two_factor_authentication_controller.rb

```diff
@@ -11,10 +11,15 @@ class Devise::TwoFactorAuthenticationController < DeviseController
   def new
     current_user.send_new_otp
     flash[:notice] = t('devise.two_factor_authentication.user.new_otp_sent')
-    redirect_to user_two_factor_authentication_path
+    redirect_to user_two_factor_authentication_path(method: 'sms')
```

Comment:
Could you please explain why the method parameter is needed? It seems like the `use_totp?` logic should be enough to determine whether or not to show the `:show_totp` view. When I removed the `&& params[:method] != 'sms'` from the `use_totp?` logic, all tests passed except for `spec/controllers/devise/two_factor_authentication_controller_spec.rb:216`, which is expected since I changed the logic.